File: src-tauri/src/metrics/datapoint.rs
use plotters::style::RGBColor;

#[derive(Debug, Clone)]
pub struct DataPoint {
    pub identifier: u32,
    pub size: f32,
    pub stroke_width: u32,
    pub color: RGBColor,
    pub x: f32,
    pub y: f32,
}

impl DataPoint {
    pub fn new(identifier: &u32, size: &f32, stroke_width: &u32, color: &RGBColor, x: &f32, y: &f32) -> DataPoint { 
        return DataPoint { identifier: *identifier, x: *x, y: *y , size: *size, stroke_width:*stroke_width, color: *color};
    }

    pub fn toTuple() -> (f32, f32) {
        todo!()
    }
}

File: src/app/models/datapoint.ts
export class DataPoint{
    identifier: number;
    size: number;
    pattern_size: number;
    density: number;
    stroke_width: number;

    x: number;
    y: number;

    r: number;
    g: number;
    b: number;
    a: number;

    constructor(identifier: number, size: number,  pattern_size: number, density: number, stroke_width: number, x: number, y: number, r: number, g: number, b: number, a:number){
        this.identifier = identifier;
        this.pattern_size = pattern_size;
        this.size = size;
        this.density = density;
        this.stroke_width = stroke_width;
        this.x = x;
        this.y = y;
        this.r = r;
        this.g = g;
        this.b = b;
        this.a = a;
    }
}